firebase-common/src/main/java/com/google/firebase/Timestamp.kt

```diff
@@ -69,12 +69,16 @@ class Timestamp : Comparable<Timestamp>, Parcelable {
   }
 
   /**
-   * Returns a new Date corresponding to this timestamp.
+   * Returns a new [Date] corresponding to this timestamp.
    *
    * This may lose precision.
    */
   fun toDate(): Date = Date(seconds * 1_000 + (nanoseconds / 1_000_000))
 
+  /** Returns a new [Instant] that matches the time defined by this timestamp. */
+  @RequiresApi(Build.VERSION_CODES.O)
```

Comment:
Can you also add a way to create a Timestamp from an Instant?

Reply:
There already is one, as a constructor. Although, we could move it to a factor method if you'd prefer (which is more Kotlin like anyhow). I only added it as a constructor to align with the existing api.

Reply:
Ahh I see. I missed that. A secondary constructor is fine, especially since there are other secondary constructors.